espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```diff
@@ -0,0 +1,216 @@
+        MethodWithArguments method = findCompatibleMethod(object.getClass());
+        if (method == null) {
+            ArrayList<String> noMethodMessage = new ArrayList<>();
+            noMethodMessage.add(String.format("No such method found: %s(", InvocationOperation.this.methodName));
+
+            for (Object argument : InvocationOperation.this.arguments) {
+                String simpleName = argument == null ? "null" : argument.getClass().getSimpleName();
+                noMethodMessage.add(String.format("[%s]", simpleName));
+            }
+            noMethodMessage.add(")");
+
+            logger.debug("Failed to find a suitable method using reflection");
+            throw new AppiumException(BackdoorResultVoid.instance.asMap(InvocationOperation.this.methodName, object, TextUtils.join("", noMethodMessage)).toString());
+
+        }
+        return method;
+    }
+
+    @Nullable
+    public MethodWithArguments findCompatibleMethod(Class<?> forClass) {
+        // Fast path
+        try {
+            return new MethodWithArguments(forClass.getMethod(methodName, argumentTypes),
+                    new ArrayList<Object>(arguments));
+        } catch (NoSuchMethodException e) {
+            // No immediate method found
+
+            for (Method method : forClass.getMethods()) {
+                if (!method.getName().equals(methodName)) continue;
+
+                try {
+                    return new MethodWithArguments(method, parseToSuitableArguments(method));
+                } catch (IncompatibleArgumentsException e1) {
+                    continue;
+                }
+            }
+
+            return null;
+        }
+    }
+
+    private List<Object> parseToSuitableArguments(Method method) throws IncompatibleArgumentsException {
+        List<Object> suitableArguments = new ArrayList<Object>(arguments.size());
+
+        if (arguments.size() != method.getParameterTypes().length)
+            throw new IncompatibleArgumentsException(String.format("Argument count mismatch. Given:'%d' - Possibility:'%d'", arguments.size(), method.getParameterTypes().length));
+
+        for (int i = 0; i < method.getParameterTypes().length; i++) {
+            Object argument = arguments.get(i);
+            Class<?> parameterType = method.getParameterTypes()[i];
+            suitableArguments.add(parseToSuitableArgument(argument, parameterType));
+        }
+
+        return suitableArguments;
+    }
+
+    private Object parseToSuitableArgument(Object argument, Class<?> parameterType) throws IncompatibleArgumentsException {
+        if (argument == null) {
+            if (!parameterType.isPrimitive()) return null;
+            throw new IncompatibleArgumentsException(String.format("Null is incompatible with primitive '%s'", parameterType));
+        }
+
+        if (argument.getClass() == parameterType) return argument;
+        if (BackdoorUtils.mapToPrimitiveClass(argument.getClass()) == parameterType)
+            return argument;
```

Comment:
Please always add curly brackets unless you have single return in the same line where if is present